### ISSUE TITLE
qutebrowser completely freezes for 25s when libnotify daemon is unreachable via DBus

### ISSUE THREAD

**Issue by s3lph** [qutebrowser/qutebrowser#7017]
**Version info**:
qutebrowser v2.4.0
Git commit: 
Backend: QtWebEngine 5.15.8, based on Chromium 87.0.4280.144
Qt: 5.15.2

**Does the bug happen if you start with `--temp-basedir`?**:
Yes
**Description**
When `content.notifications.presenter` is set to `libnotify`, but the notification daemon is currently not running, qutebrowser will lock up completely for 25s, then recover and afterwards print the follwing message to stdout:

```
17:19:29 ERROR: Failed to initialize libnotify notification adapter: Could not construct a DBus interface: org.freedesktop.DBus.Error.NoReply - Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
```

The [25s seem to come from qdbus](https://doc.qt.io/qt-5/qdbusconnection.html#call), and the qdbus call in questions seems to be at 
https://github.com/qutebrowser/qutebrowser/blob/bb265dc712b56913442079a20a494cfd1079d58a/qutebrowser/browser/webengine/notification.py#L981-L992

**How to reproduce**

1. Stop your notification daemon (`systemctl --user stop mako.service` in my case)
2. Visit any site that shows a notification
3. Wait 25s until qutebrowser becomes responsive again.

**Comment by The-Compiler** [comment]
Right, 25s seems overly long. Don't think I agree with "implementation-defined value that is suitable for inter-process communications (generally, 25 seconds)". Something like 5s or so might be more appropriate. Though ideally we shouldn't run into a timeout at all, and I thought we did not. Will try to reproduce later.

**Comment by s3lph** [comment]
The timeout is one issue, however completely apart from this: Even if this blocks due to an unreachable notification daemon, something like this should never lock up the entire browser.

**Comment by The-Compiler** [comment]
Hence "Though ideally we shouldn't run into a timeout at all, and I thought we did not.". I agree we shouldn't hang at all, at least if this is a common occurrence. If some other component misbehaves, I don't think it's worth the additional complexity in qutebrowser to avoid a (reasonably short) hang by all costs.

I can reproduce - however, I believe this is something new, perhaps triggered by a change in DBus or something. For starters, this is the first time this comes up, almost a year after notification support was released. But I also remember that when I originally implemented this, systemd/DBus took care of starting the notification daemon when a notification comes in. In my case with `dunst`, I believe this is either done by the `BusName` in `/usr/lib/systemd/user/dunst.service`:

```ini
[Unit]
Description=Dunst notification daemon
Documentation=man:dunst(1)
PartOf=graphical-session.target

[Service]
Type=dbus
BusName=org.freedesktop.Notifications
ExecStart=/usr/bin/dunst
```

and/or the `Exec` in `/usr/share/dbus-1/services/org.knopwob.dunst.service`:

```ini
[D-BUS Service]
Name=org.freedesktop.Notifications
Exec=/usr/bin/dunst
SystemdService=dunst.service
```

however, it looks like that doesn't work reliably anymore. After the timeout, I can see this in `journalctl`:

```
Feb 12 09:12:25 aragog plasma_waitforname[299152]: org.kde.knotifications: WaitForName: Service was not registered within timeout
Feb 12 09:12:25 aragog dbus-daemon[1939]: [session uid=1000 pid=1939] Activated service 'org.freedesktop.Notifications' failed: Process org.freedesktop.Notifications exited with status 1
Feb 12 09:12:45 aragog dbus-daemon[1939]: [session uid=1000 pid=1939] Activating service name='org.freedesktop.Notifications' requested by ':1.26632' (uid=1000 pid=299256 comm="dbus-send --print-reply=literal --dest=org.freedes")
```

do you see something similar too?

**Comment by s3lph** [comment]
> Hence "Though ideally we shouldn't run into a timeout at all, and I thought we did not.". I agree we shouldn't hang at all, at least if this is a common occurrence.

Ah ok, then I misunderstood you here.

> do you see something similar too?

In my situation I had misconfigured my notification daemon, so even when it attempted to start, the unit would just fail and the DBus service wasn't becoming available.

**Comment by The-Compiler** [comment]
On second though, I wonder if there is a bug lurking in Qt or PyQt there as well: Our [call](https://github.com/qutebrowser/qutebrowser/blob/bb265dc712b56913442079a20a494cfd1079d58a/qutebrowser/browser/webengine/notification.py#L981-L992) uses `QDBus.BlockWithGui`, which [is documented](https://doc.qt.io/qt-5/qdbus.html#CallMode-enum) as:

> Use the Qt event loop to wait for a reply. This means that the user-interface will stay responsive (processing input events), but it also means other events may happen, like signal delivery and other D-Bus method calls.

...yet clearly that's not what's happening. No [matching Qt bugreports](https://bugreports.qt.io/browse/QTBUG-6393?jql=text%20~%20BlockWithGui), from what I can see.
